department-of-veterans-affairs/caseflow
# Piedram/appeals 4361

Resolves [APPEAL-4361](https://vajira.max.gov/browse/APPEALS-4361)

### Description
After a successful split appeal post to the database, a success banner will display on the queue page to show the user that the appeal has been successfully split. If the post fails for whatever reason, the banner will display a failure message so the user knows the split appeal was not created. 

### Acceptance Criteria

- [ ] When the user selects "Split Appeal" on the review and confirm new appeal form, the user is directed to the Case Details screen

#### If the split was successful a success banner appears 

- [ ] Banner Header: You have successfully split <appellant>'s appeal

- [ ] <appellant> is the appellant's name

- [ ] Banner text: This new appeal stream has the same docket number and tasks as the original appeal.

- [ ] If the split did not process correctly (e.g., server issue) then the user is informed the split was not successful

### Testing Plan
1. Go to localhost:3000 and sign in as an SSC user.
![image](https://user-images.githubusercontent.com/99915461/186937504-7f019df5-c3d9-4cb5-9d43-d713f5a72eb7.png)

2. Go to Queue via **your Queue** link, go to search at the top and enter a user id. When a user pulls up, click the **Docket** link to navigate to the docket page.
![image](https://user-images.githubusercontent.com/99915461/186937698-8f0de20b-1eb2-4176-9532-34a7cb75ef6c.png)
![image](https://user-images.githubusercontent.com/99915461/186937862-8f1368b6-1ffd-4ca5-90c1-c6bc78e5abc4.png)

3. Click the **Correct issues** link to navigate to the edit issues page. 
![image](https://user-images.githubusercontent.com/99915461/186938581-f6bf7ba2-6c2e-4fbf-9c9d-4d3b507d5d0a.png)

4. If the appeal doesn't have any issues, assign at least 2 issues to the appeal by using the **add issue** button. Click save to save the issues to the appeal.
![image](https://user-images.githubusercontent.com/99915461/186938795-26316a08-e3ec-447d-be00-626d16a2d5cf.png)
![image](https://user-images.githubusercontent.com/99915461/186938882-53a06946-ace4-4bc8-924f-9a6d0cdb1ac0.png)
![image](https://user-images.githubusercontent.com/99915461/186939000-c38f1ddb-d6f8-47a8-b90e-86bf0cb3e87f.png)
![image](https://user-images.githubusercontent.com/99915461/186939025-59d8520b-d0be-4a73-9c49-0ac4e65de93b.png)

5. Click the **Correct issues** link again and then click **Split appeal** to navigate to the split appeal page.
![image](https://user-images.githubusercontent.com/99915461/186939215-ca8aeedb-86a4-4740-a6c6-7171c3767ad1.png)
![image](https://user-images.githubusercontent.com/99915461/186939252-ff0f34de-b6a8-4c27-9712-8328ec470811.png)

6. Select a reason for splitting the appeal and one of the issues. Then click **continue**.
![image](https://user-images.githubusercontent.com/99915461/186939364-63d5722b-c0fa-4341-92e9-610aa4b719f2.png)
![image](https://user-images.githubusercontent.com/99915461/186939427-a930c6d0-7a8b-43c0-b52f-25841d0449b0.png)

7. On the **Review** page, click **Split appeal**.
![image](https://user-images.githubusercontent.com/99915461/186939554-687f430b-caea-4cae-bf06-2661c2c98f02.png)

8. The success banner will display with the user's name. 
![image](https://user-images.githubusercontent.com/99915461/186939648-d56e6838-2bff-48d2-acdd-1be22bac00d2.png)

9. To make sure the banner disappears on a new page load, refresh the page. The banner should disappear. 
![image](https://user-images.githubusercontent.com/99915461/186942360-93d37650-1651-4bfa-858e-8ee7d0edc7a0.png)


#### To Test the Failure banner.
1. Go into the post function for the Split Appeal button. The function is located at **client/app/intake/actions/intake.js**. On **line 32**, add any value between the closing bracket **}** of **${ENDPOINT_NAMES.SPLIT_APPEAL}** and the backtick **`** like the example below. Save the changes to test them.
![image](https://user-images.githubusercontent.com/99915461/186940514-ae22ca78-5939-4525-b272-45dff121bc93.png)

2. Open the case details page from step 9 and click the back arrow to return to the **review split** page. 
![image](https://user-images.githubusercontent.com/99915461/186943271-afee2d95-8ad1-4a3e-ba48-b7b5ac06847c.png)
![image](https://user-images.githubusercontent.com/99915461/186943407-5e1668c3-37ac-422e-826c-c7b241db0c5a.png)

3. Click **Split appeal**. The failure banner should render from the post failure. 
![image](https://user-images.githubusercontent.com/99915461/186943553-cadf9a3e-b868-4de2-aea6-583ab15c83f3.png)

4. Refresh the page to make sure the banner doesn't re-render. 
![image](https://user-images.githubusercontent.com/99915461/186943643-7bdd15b9-f86c-4257-9705-0ae9c15b6524.png)

5. Revert the code changes in the post function and save. 
![image](https://user-images.githubusercontent.com/99915461/186943874-e1bff823-552d-48f5-b6b9-5b6824123f2f.png)

6. If you want to make sure the post was reverted correctly, repeat steps 2-4 and make sure the banner shows properly.
![image](https://user-images.githubusercontent.com/99915461/186944299-0eca97ae-7a2f-4c0a-87e9-ea5378c10adc.png)

 BEFORE|AFTER

#### Before 
![image](https://user-images.githubusercontent.com/99915461/186943643-7bdd15b9-f86c-4257-9705-0ae9c15b6524.png)
#### After 
Success
![image](https://user-images.githubusercontent.com/99915461/186939648-d56e6838-2bff-48d2-acdd-1be22bac00d2.png)
Failure
![image](https://user-images.githubusercontent.com/99915461/186943553-cadf9a3e-b868-4de2-aea6-583ab15c83f3.png)
